holopy/core/tests/test_metadata.py

```diff
@@ -0,0 +1,666 @@
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+        self.assertEqual(points.r.size, npts)
+        self.assertEqual(points.theta.size, npts)
+        self.assertEqual(points.phi.size, npts)
+
+    @attr("fast")
+    def test_stores_rthetaphi_correct_values_when_rthetaphi_passed(self):
+        np.random.seed(70)
+        npts = 21
+        r = np.random.randn(npts)
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+        self.assertTrue(np.all(points.r == r))
+        self.assertTrue(np.all(points.theta == theta))
+        self.assertTrue(np.all(points.phi == phi))
+
+    @attr("fast")
+    def test_stores_r_as_array_when_scalar_r_passed(self):
+        np.random.seed(70)
+        npts = 21
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+        # Then we pick a scalar r:
+        r = np.random.randn(1).squeeze()
+
+        points = detector_points(r=r, theta=theta, phi=phi)
+
+        self.assertTrue(np.all(points.r == r))
+        self.assertEqual(points.r.size, npts)
+
+    @attr("fast")
+    def test_r_defaults_to_inf_when_thetaphi_passed(self):
+        np.random.seed(70)
+        npts = 21
+        theta = np.random.randn(npts) % np.pi
+        phi = np.random.randn(npts) % (2 * np.pi)
+
+        points = detector_points(theta=theta, phi=phi)
+
+        self.assertTrue(np.all(np.isinf(points.r)))
+        self.assertEqual(points.r.size, npts)
+
+    # Other tests:
+    @attr("fast")
+    def test_data_is_stored_as_zeros_of_corect_size(self):
+        npts = 23
+        x, y, z = np.random.randn(3, npts)
+        points = detector_points(x=x, y=y, z=z)
+        self.assertEqual(points.size, npts)
+        self.assertTrue(np.all(points.values == 0))
+
+    @attr("fast")
+    def test_name_defaults_to_data(self):
+        x, y, z = np.random.randn(3, 10)
+        points = detector_points(x=x, y=y, z=z)
+        self.assertEqual(points.name, 'data')
+
+    @attr("fast")
+    def test_name_is_stored(self):
+        x, y, z = np.random.randn(3, 10)
+        name = 'this-is-a-test'
+        points = detector_points(x=x, y=y, z=z, name=name)
+        self.assertEqual(points.name, name)
+
+    # FIXME no checks for normal default values
+
+    @attr("fast")
+    def test_has_attribute_normals(self):
+        x, y, z = np.random.randn(3, 10)
+        points = detector_points(x=x, y=y, z=z)
+        self.assertTrue(hasattr(points, 'normals'))
+
+    @attr("fast")
+    def test_default_normals_are_shape_3xN_for_spherical_coords(self):
+        npts = 13
+        r, theta, phi = np.random.randn(3, npts)
+        points = detector_points(r=r, theta=theta, phi=phi)
+        self.assertEqual(points.normals.values.shape, (3, npts))
+
+    @attr("fast")
+    @unittest.skip("Fails. Not sure if test is wrong or code")
```

Comment:
3. This test currently fails, as specifying individual normals for ```detector_points``` is not supported. I feel like this is something we might want. Thoughts? I could leave this as a failing test and mention it as part of a (new) refactor issue?

Reply:
It would be nice to raise an error if you try to pass in normals instead.

Reply:
Should be resolved by 503325f